### PR TITLE
Remember map size selection

### LINKED ISSUE #3260: Scenario selection window, scenario map size arrangement should be memorized
image
Map size arrangement should be saved, when we choose scenario and then return to the list back. Currently, in fheroes2 each time we enter scenario list, upper "map size" buttons never save their state.
Also, in the OG these buttons save their state even if we start game and then return back to main menu.

### PR DESCRIPTION
**Pull request by ihhub**
close #3260

### CODE REVIEW

File: src/fheroes2/dialog/dialog_selectscenario.cpp [hunk 1]
```diff
@@ -334,8 +350,47 @@ const Maps::FileInfo * Dialog::SelectScenario( const MapsFileInfoList & all, siz
     fheroes2::Button buttonSelectXLarge( rt.x + 223, rt.y + 22, ICN::REQUESTS, 15, 16 );
     fheroes2::Button buttonSelectAll( rt.x + 285, rt.y + 22, ICN::REQUESTS, 17, 18 );
 
-    buttonSelectAll.press();
-    fheroes2::ButtonBase * currentPressedButton = &buttonSelectAll;
+    fheroes2::ButtonBase * currentPressedButton = nullptr;
+
+    // This variable is used to remember the selection of map size throught the game.
```
Comment (oleg-derevenetz) [inline src/fheroes2/dialog/dialog_selectscenario.cpp:355]: I suppose there is a typo: `throught` -> `through` :)

File: src/fheroes2/dialog/dialog_selectscenario.cpp [hunk 2]
```diff
@@ -359,8 +414,33 @@ const Maps::FileInfo * Dialog::SelectScenario( const MapsFileInfoList & all, siz
...
     listbox.SetAreaMaxItems( 9 );
-    listbox.SetAreaItems( fheroes2::Rect( rt.x + 55, rt.y + 55, 270, 175 ) );
-    listbox.SetListContent( const_cast<MapsFileInfoList &>( all ) );
+    listbox.SetAreaItems( { rt.x + 55, rt.y + 55, 270, 175 } );
+
+    size_t selectedId = 0;
+
+    switch ( selectedMapSize ) {
+    case Maps::SMALL:
+        listbox.SetListContent( const_cast<MapsFileInfoList &>( small ) );
```
Comment (oleg-derevenetz) [inline src/fheroes2/dialog/dialog_selectscenario.cpp:423]: Are you sure we need `const_cast` here and in other similar places? `all` is the reference to const, but `small`, `medium`, etc are not.

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @ihhub I left a few suggestions here, please take a look when it's convenient for you.

File: src/fheroes2/dialog/dialog_selectscenario.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_selectscenario.cpp]: You're right!

File: src/fheroes2/dialog/dialog_selectscenario.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_selectscenario.cpp]: Yup, copy-paste mistake.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**